advanced/Scripts/version.sh

```diff
@@ -8,67 +8,98 @@
 # This file is copyright under the latest version of the EUPL.
 # Please see LICENSE file for your rights under this license.
 
+# Variables
+DEFAULT="-1"
+PHGITDIR="/etc/.pihole/"
+WEBGITDIR="/var/www/html/admin/"
 
+getLocalVersion() {
+  # Get the tagged version of the local repository
+  local directory="${1}"
+  local version
 
-# Flags:
-latest=false
-current=false
+  cd "${directory}" || { echo "${DEFAULT}"; return 1; }
+  version=$(git describe --tags --always || \
+            echo "${DEFAULT}")
+  if [[ "${version}" =~ ^v ]]; then
+    echo "${version}"
+  elif [[ "${version}" == "${DEFAULT}" ]]; then
+    echo "ERROR"
+    return 1
+  else
+    echo "Untagged"
+  fi
+  return 0
+}
 
-DEFAULT="-1"
+getLocalHash() {
+  # Get the short hash of the local repository
+  local directory="${1}"
+  local hash
 
-normalOutput() {
-	piholeVersion=$(cd /etc/.pihole/ && git describe --tags --abbrev=0)
-	webVersion=$(cd /var/www/html/admin/ && git describe --tags --abbrev=0)
+  cd "${directory}" || { echo "${DEFAULT}"; return 1; }
+  hash=$(git rev-parse --short HEAD || \
+         echo "${DEFAULT}")
+  if [[ "${hash}" == "${DEFAULT}" ]]; then
+    echo "ERROR"
+    return 1
+  else
+    echo "${hash}"
+  fi
+  return 0
+}
+
+getRemoteVersion(){
+  # Get the version from the remote origin
+  local daemon="${1}"
+  local version
 
-	piholeVersionLatest=$(curl -s https://api.github.com/repos/pi-hole/pi-hole/releases/latest | grep -Po '"tag_name":.*?[^\\]",' |  perl -pe 's/"tag_name": "//; s/^"//; s/",$//')
-	webVersionLatest=$(curl -s https://api.github.com/repos/pi-hole/AdminLTE/releases/latest | grep -Po '"tag_name":.*?[^\\]",' |  perl -pe 's/"tag_name": "//; s/^"//; s/",$//')
+  version=$(curl --silent --fail https://api.github.com/repos/pi-hole/${daemon}/releases/latest | \
```

Comment:
I wonder if we should store the most recent tag somewhere on our servers, so that there is (a) no limit on how often you can query the latest version and (b) no effort necessary to parse it on the client side